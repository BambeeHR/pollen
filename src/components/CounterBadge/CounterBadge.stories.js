import { number } from '@storybook/addon-knobs';
import CounterBadge, { Sizes, Variants } from './CounterBadge.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

export default {
  title: 'CounterBadge',
  components: CounterBadge,
};

const sizes = Object.values(Sizes);
const variants = Object.values(Variants);

export const Gallery = () => ({
  components: { CounterBadge, TypeOverline },
  props: {
    value: {
      default: number('Value', 8),
    },
  },
  data() {
    return { sizes, variants };
  },
  template: `
  <div>
    <div v-for="variant in variants" :key="variant" class="mb-8">
      <TypeOverline tag="h1" variant="large" class="mb-2">{{ variant }}</TypeOverline>
      <div class="flex">
        <div v-for="size in sizes" class="mr-4">
          <TypeOverline tag="h2" class="mb-2">{{ size }}</TypeOverline>
          <CounterBadge :value="value" :size="size" :variant="variant" />
        </div>
      </div>
    </div>
  </div>
  `,
});
