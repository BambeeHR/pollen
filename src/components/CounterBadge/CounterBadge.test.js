import { mount } from '@vue/test-utils';
import CounterBadge from './CounterBadge.vue';
import { Gallery } from './CounterBadge.stories';

describe('CounterBadge', () => {
  test('renders correctly', () => {
    const wrapper = mount(CounterBadge, {
      propsData: {
        value: 8,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders gallery correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('rounds large numbers', () => {
    const wrapper = mount(CounterBadge, {
      propsData: {
        value: 1200,
      },
    });
    expect(wrapper.text()).toBe('1.2K');
  });
});
