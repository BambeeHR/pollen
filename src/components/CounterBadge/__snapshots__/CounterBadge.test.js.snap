// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterBadge renders correctly 1`] = `
<div
  class="counter-badge counter-badge--medium counter-badge--primary"
>
  
  8

</div>
`;

exports[`CounterBadge renders gallery correctly 1`] = `
<div>
  <div
    class="mb-8"
  >
    <h1
      class="type-overline type-overline-large mb-2"
    >
      primary
    </h1>
     
    <div
      class="flex"
    >
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          small
        </h2>
         
        <div
          class="counter-badge counter-badge--small counter-badge--primary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          medium
        </h2>
         
        <div
          class="counter-badge counter-badge--medium counter-badge--primary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          large
        </h2>
         
        <div
          class="counter-badge counter-badge--large counter-badge--primary"
        >
          
  8

        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-8"
  >
    <h1
      class="type-overline type-overline-large mb-2"
    >
      secondary
    </h1>
     
    <div
      class="flex"
    >
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          small
        </h2>
         
        <div
          class="counter-badge counter-badge--small counter-badge--secondary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          medium
        </h2>
         
        <div
          class="counter-badge counter-badge--medium counter-badge--secondary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          large
        </h2>
         
        <div
          class="counter-badge counter-badge--large counter-badge--secondary"
        >
          
  8

        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-8"
  >
    <h1
      class="type-overline type-overline-large mb-2"
    >
      tertiary
    </h1>
     
    <div
      class="flex"
    >
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          small
        </h2>
         
        <div
          class="counter-badge counter-badge--small counter-badge--tertiary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          medium
        </h2>
         
        <div
          class="counter-badge counter-badge--medium counter-badge--tertiary"
        >
          
  8

        </div>
      </div>
      <div
        class="mr-4"
      >
        <h2
          class="type-overline type-overline-small mb-2"
        >
          large
        </h2>
         
        <div
          class="counter-badge counter-badge--large counter-badge--tertiary"
        >
          
  8

        </div>
      </div>
    </div>
  </div>
</div>
`;
